### ISSUE TITLE
[3.1.0] Cannot assign to read only property 'form' of object

### ISSUE THREAD

**Issue by Dhaulagiri** [emberjs/ember.js#16385]
In the latest 3.1 beta, if I have a button compoment like the below, any attempt to bind to the `form` attribute triggers this error:

```
Uncaught TypeError: Cannot assign to read only property 'form' of object '#<HTMLButtonElement>'
    at NewElementBuilder.__setProperty (vendor.js:29024)
    at DefaultDynamicProperty.set (vendor.js:28014)
    at NewElementBuilder.setDynamicAttribute (vendor.js:29035)
    at ComponentElementOperations.flush (vendor.js:27024)
    at Object.evaluate (vendor.js:26541)
    at AppendOpcodes.evaluate (vendor.js:25357)
    at LowLevelVM.evaluateSyscall (vendor.js:28562)
    at LowLevelVM.evaluateInner (vendor.js:28538)
    at LowLevelVM.evaluateOuter (vendor.js:28530)
    at VM.next (vendor.js:30113)
```

```js
import Component from '@ember/component';

export default Component.extend({
  tagName: 'button',
  attributeBindings: ['form']
});
```

```hbs
{{my-button form="my-form"}}
```

This does not occur in 3.0

_I would make a twiddle to reproduce but I can't seem to remember how to point a twiddle at the beta 🤔_

Possibly related to the glimmer-vm upgrade in https://github.com/emberjs/ember.js/pull/15828?

**Comment by Dhaulagiri** [comment]
This is still an issue in the 3.1 release version

**Comment by rwjblue** [comment]
@Dhaulagiri - Can you throw together a demo app (`ember new foo` + tweak to show the error)?

**Comment by Dhaulagiri** [comment]
@rwjblue if you pull down https://github.com/Dhaulagiri/form-demo and run you will see the error

**Comment by Exelord** [comment]
Same issue here :/

**Comment by Dhaulagiri** [comment]
@Exelord have you had any success finding a workaround for this?

**Comment by Exelord** [comment]
Nope :/ I'm just holding on with the upgrade.

**Comment by rwjblue** [comment]
The issue is almost certainly in glimmer-vm itself. I think the next step is to make a failing test PR over there. Based on your demo, I think it would basically be similar to [this test](https://github.com/glimmerjs/glimmer-vm/blob/a9fb9ebe35df3e01dfc9b8672aa5cb1d22f00ecf/packages/%40glimmer/runtime/test/attributes-test.ts#L150-L154)...

**Comment by Dhaulagiri** [comment]
@Exelord you can get the fix for this in 3.1 by pointing your `ember-source` to `https://s3.amazonaws.com/builds.emberjs.com/release/shas/4cb565e3924363b165736cbaff8e4ee5d29218d7.tgz` in your package.json 👌

**Comment by Exelord** [comment]
Thank you @Dhaulagiri for your work on it! Can't wait for official release :)
